plugins/inputs/zfs/zfs_freebsd.go

```diff
@@ -87,6 +87,61 @@ func (z *Zfs) gatherPoolStats(acc telegraf.Accumulator) (string, error) {
 	return strings.Join(pools, "::"), nil
 }
 
+func (z *Zfs) gatherDatasetStats(acc telegraf.Accumulator) (string, error) {
+
+	lines, err := z.zdataset()
+	if err != nil {
+		return "", err
+	}
+
+	datasets := []string{}
+	for _, line := range lines {
+		col := strings.Split(line, "\t")
+
+		datasets = append(datasets, col[0])
+	}
+
+	if z.DatasetMetrics {
+		for _, line := range lines {
+			col := strings.Split(line, "\t")
+			if len(col) != 5 {
+				continue
+			}
```

Comment:
With the `properties` definition above we can make this easily extensible via
```suggestion
			if len(col) != len(properties) {
				continue
			}
```